src/useFirebaseUser.js

```diff
@@ -43,28 +43,53 @@ const defaultTokenChangedHandler = async (authUser) => {
   return response
 }
 
-const setAuthCookie = async (firebaseUser) => {
+/**
+ * Fetch the currentusers idTokenResult which contains both the idToken and the claims
+ * https://firebase.google.com/docs/reference/js/firebase.auth.IDTokenResult
+ */
+const fetchIdTokenResult = async () => {
+  try {
+    const idTokenResult = await firebase.auth().currentUser.getIdTokenResult()
+    return idTokenResult
+  } catch (e) {
+    logDebug('Error Getting Id Token Result', e)
+    return null
+  }
+}
+
+const setAuthCookie = async (firebaseUser, token) => {
   const { tokenChangedHandler } = getConfig()
+
   const authUser = createAuthUser({
     firebaseUserClientSDK: firebaseUser,
     clientInitialized: true,
   })
 
   if (tokenChangedHandler) {
-    return tokenChangedHandler(authUser)
+    return tokenChangedHandler(authUser, token)
   }
 
-  return defaultTokenChangedHandler(authUser)
+  return defaultTokenChangedHandler(authUser, token)
 }
 
 const useFirebaseUser = () => {
   const [user, setUser] = useState()
   const [initialized, setInitialized] = useState(false)
 
   async function onIdTokenChange(firebaseUser) {
-    setUser(firebaseUser)
+    let idTokenResult = { claims: null, token: null }
```

Comment:
Make sure claims default to an empty object. Either set to `undefined` here (relying on defaults in `createAuthUser`) or `{}`.